x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.ts

```diff
@@ -42,11 +43,7 @@ export async function fetchIndexShardSize(
       query: {
         bool: {
           must: [
-            {
-              match: {
-                type: 'index_stats',
-              },
-            },
+            createDatasetFilter('index_stats', 'elasticsearch.index'),
```

Comment:
@weltenwort I used the new createDatasetFilter function here though it's not a filter.  I'm not sure why this particular alert query is so different from the others and whether its necessary.  Can you have a look and see if you see anything that would not work with my changes.  The query would look like this:

```
    "query": {
      "bool": {
        "must": [
          {
            "bool": {
              "should": [
                {
                  "term": {
                    "type": "index_stats"
                  }
                },
                {
                  "term": {
                    "data_stream.dataset": "elasticsearch.index"
                  }
                }
              ],
              "minimum_should_match": 1
            }
          },
          {
            "range": {
              "timestamp": {
                "gte": "now-5m"
              }
            }
          }
        ]
      }
    },


```

Reply:
I can't think of a justification to use `match` here. The field is a `keyword` and we're looking for an exact match, so a `term` makes more sense.

Regarding the outer `bool`, it's also not clear why this is a `must` instead of a `filter`. The latter would be cachable and we're not using the score anyway (with `size` being `0`). Do we want to improve that while we're at it?

Reply:
Thanks!  Done.